src/Cli/dotnet/commands/dotnet-nuget/NuGetCommandParser.cs

```diff
@@ -120,22 +120,70 @@ private static CliCommand GetTrustCommand()
             CliCommand trustCommand = new("trust");
 
-            CliArgument<string> commandArgument = new CliArgument<string>("command") { Arity = ArgumentArity.ZeroOrOne };
-            commandArgument.AcceptOnlyFromAmong(new string[] { "list", "author", "repository", "source", "certificate", "remove", "sync" });
+            CliOption<bool> allowUntrustedRoot = new("--allow-untrusted-root");
+            CliOption<string> owners = new("--owners");
 
-            trustCommand.Arguments.Add(commandArgument);
+            trustCommand.Subcommands.Add (new CliCommand("list"));
+            trustCommand.Subcommands.Add (AuthorCommand());
+            trustCommand.Subcommands.Add (RepositoryCommand());
+            trustCommand.Subcommands.Add (SourceCommand());
+            trustCommand.Subcommands.Add (CertificateCommand());
+            trustCommand.Subcommands.Add (RemoveCommand());
+            trustCommand.Subcommands.Add (SyncCommand());
 
-            trustCommand.Options.Add(new CliOption<string>("--algorithm"));
-            trustCommand.Options.Add(new CliOption<bool>("--allow-untrusted-root"));
-            trustCommand.Options.Add(new CliOption<string>("--owners"));
-            trustCommand.Options.Add(new CliOption<string>("--configfile"));
-            trustCommand.Options.Add(CommonOptions.VerbosityOption);
+            CliOption<string> configFile = new("--configfile");
+
+            // now set global options for all nuget commands: configfile, verbosity
+            // as well as the standard NugetCommand.Run handler
 
+            trustCommand.Options.Add(configFile);
+            trustCommand.Options.Add(CommonOptions.VerbosityOption);
             trustCommand.SetAction(NuGetCommand.Run);
 
+            foreach (var command in trustCommand.Subcommands)
+            {
+                command.Options.Add(configFile);
+                command.Options.Add(CommonOptions.VerbosityOption);
+                command.SetAction(NuGetCommand.Run);
+            }
+
+            CliCommand AuthorCommand() => new CliCommand("author") {
+                new CliArgument<string>("NAME"),
+                new CliArgument<string>("PACKAGE"),
+                allowUntrustedRoot,
+            };
+
+            CliCommand RepositoryCommand() => new CliCommand("repository") {
+                new CliArgument<string>("NAME"),
+                new CliArgument<string>("PACKAGE"),
+                allowUntrustedRoot,
+                owners
+            };
+
+            CliCommand SourceCommand() => new CliCommand("source") {
+                new CliArgument<string>("NAME"),
+                owners,
+                new CliOption<string>("--source-url"),
+            };
+
+            CliCommand CertificateCommand() => new CliCommand("certificate") {
+                new CliArgument<string>("NAME"),
+                new CliArgument<string>("FINGERPRINT"),
+                allowUntrustedRoot,
+                new CliOption<string>("--algorithm")
```

Comment:
nit: I could implement `FromAmong` here - that would give tab-completion a nicer experience. But it's kind of moot because `help` doesn't come from the SDK, which is the other place that FromAmong is used.